source/common/common/cleanup.h

```diff
@@ -10,11 +10,19 @@ namespace Envoy {
 // RAII cleanup via functor.
 class Cleanup {
 public:
-  Cleanup(std::function<void()> f) : f_(std::move(f)) {}
+  Cleanup(std::function<void()> f) : f_(std::move(f)), cancelled_(false) {}
```

Comment:
If you don't end up reverting this change because of the TODO we discussed, can you add an explicit test for this somewhere in common tests in a follow up? If we keep this code I might also use an absl::optional for the functor instead of an explicit cancelled_ boolean, but that's a minor thing.

Reply:
Both this and the protobuf utility had explicit tests in `test/common/common/cleanup_test.cc` and `test/common/protobuf/utility_test.cc`.

Reply:
Cool thanks, I missed that.